File: add_medicine.py
print("Python file started successfully")

from flask import Flask, request, send_file, Response
from db import get_connection

app = Flask(__name__)
@app.route("/")
def home():
   return send_file("add_medicine.html")

@app.route("/add_medicine.js")
def javascript():
    with open("add_medicine.js", "r") as file:
        return Response(file.read(), mimetype="application/javascript")
    

@app.route("/save_medicine", methods=["POST"])
def save_medicine():
    print("save_medi called")
    data = request.json

    medicine_name = data["medicine_name"]
    company_name = data["company_name"]
    category = data["category"]
    mfg_date = data["mfg_date"]
    print(mfg_date)
    exp_date = data["exp_date"]
    price = data["price"]
    description = data["description"]
    quantity = data["quantity"]
    batch_no = data["batch_no"]


    conn = get_connection()
    cursor = conn.cursor()
    print("DB Connected",conn)

    query = """
    INSERT INTO medicines
    (
        medicine_name,
        company_name,
        category,
        manufacturing_date,
        expiry_date,
        price,
        description,
        quantity,
        batch_no
    )
    VALUES
    (%s,%s,%s,%s,%s,%s,%s,%s,%s)
    """
    values = (
        medicine_name,
        company_name,
        category,
        mfg_date,
        exp_date,
        price,
        description,
        quantity,
        batch_no
    )
    cursor.execute(query, values)
    conn.commit()

    cursor.close()
    conn.close()

    return "Medicine Saved Successfully"


if __name__ == "__main__":
    app.run(debug=True)

File: add_medicine.js
function save_medicine(){
    const medicineName = document.getElementById("medicine_name").value;
    const companyName = document.getElementById("company_name").value;
    const category = document.getElementById("category").value;


const mfgDate = document.getElementById("mfg_date").value;
alert(mfg_date);
const expDate = document.getElementById("exp_date").value;
const price = document.getElementById("price").value;
const description = document.getElementById("description").value;
const quantity = document.getElementById("quantity").value;
const batchNo = document.getElementById("batch_no").value;
     let medicine = {
        medicine_name: medicineName,
        company_name: companyName,
        category: category,
        mfg_date: mfgDate,
        exp_date: expDate,
        price: price,
        description: description,
        quantity: quantity,
        batch_no: batchNo
    };
    fetch("/save_medicine", {
  //fetch("http://127.0.0.1:8000/save_medicine", {
        method: "POST",
        headers: {
            "Content-Type": "application/json"
        },
        body: JSON.stringify(medicine)
    })

    .then(response => response.text())

    .then(result => {
        alert(result);
    })

    .catch(error => {
        console.log(error);
        alert("Error saving data"+error);
    });
    alert("finished")
}



